packages/lit-dev-content/samples/examples/context-consume-provide/my-section.ts

```diff
@@ -5,30 +5,45 @@ import {levelContext} from './level-context.js';
 
 @customElement('my-section')
 export class MySection extends LitElement {
-  // Serve as a context provider with an initial value of 1.
+  // Serve as a context provider with an initial level 1 and numbering prefix
+  // based on its numbering among siblings.
   private _provider = new ContextProvider(this, {
     context: levelContext,
-    initialValue: 1,
+    initialValue: {
+      level: 1,
+      prefix: String(this._siblingNumber) + '.',
+    },
   });
 
   // Consumes level context from a parent provider. If a parent provider is
-  // found, update own provider value to be 1 greater than provided value.
+  // found, update own provider level to be 1 greater than provided value and
+  // append its sibling number.
   private _consumer = new ContextConsumer(this, {
     context: levelContext,
-    callback: (value) => {
-      this._provider.setValue(value + 1);
+    callback: ({level, prefix}) => {
+      this._provider.setValue({
+        level: level + 1,
+        prefix: prefix + String(this._siblingNumber) + '.',
+      });
     },
   });
 
+  private get _siblingNumber() {
```

Comment:
I'd be a bit weary using this over an `<ol>` even if it's for the sake of example.

Reply:
Is it possible to do this kind of numbering with `<ol>`s?
<img width="322" alt="image" src="https://github.com/lit/lit.dev/assets/40413829/b37db83b-5c22-4493-aad0-a1ec6baf2c39">

Reply:
Ah, I guess it is possible... https://stackoverflow.com/a/56430931/19498660

Reply:
AH sorry the page didn't finish rendering and I thought it was simple numbering. This looks good

Reply:
The point of this is you can do it across sections of your application though. The example doesn't show it but presumably there'll be content in between the headings and making that entire thing a `<li>` doesn't seem right? But of course the numbering can still be done with css counter so that does make this example less compelling.